DNN Platform/Library/Data/DataProvider.cs

```diff
@@ -585,6 +585,20 @@ public virtual IDataReader GetPortalSettings(int PortalId, string CultureCode)
             return ExecuteReader("GetPortalSettings", PortalId, CultureCode);
         }
 
+        internal virtual IDictionary<int, string> GetPortalSettingsBySetting(string settingName, string cultureCode)
+        {
+            var result = new Dictionary<int, string>();
+            var sql = "SELECT ps.PortalID, ps.SettingValue FROM " + DatabaseOwner + ObjectQualifier + "PortalSettings ps WHERE ps.SettingName=@0 AND COALESCE(CultureCode, @1, N'') = IsNull(@1, N'')";
```

Comment:
Do we have anywhere else in the `DataProvider` that we generate SQL like this?  Are we intentionally moving away from stored procedures?

Reply:
Also, don't we have an API method already that gets a single setting?

Reply:
My 2cts:

1. I felt we were encouraging everyone to move on from the archaic stored procedures for everything. So I kicked it off here.

2. Given that the use case for this method is not huge, I felt comfortable doing this in a way that was shortest to goal.

3. Yes, we have for a single setting. But given that an installation may have hundreds of portals, I felt it prudent to make this (which runs during a setting update) as efficient as possible.

Reply:
Using stored procedures still has a benefit, as SQL Server is able to compile it and maintain optimized query plans. This is nt important for this small query, but for many other queries in the code, i.e. imo it is not desirable to avoid sprocs and merge SQL code in other code files.